vm/systemSmartContracts/staking.go

```diff
@@ -1345,6 +1347,30 @@ func (r *stakingSC) setOwner(args *vmcommon.ContractCallInput) vmcommon.ReturnCo
 	return vmcommon.Ok
 }
 
+func (r *stakingSC) getOwner(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !r.flagSetOwner.IsSet() {
+		r.eei.AddReturnMessage("invalid method to call")
+		return vmcommon.UserError
+	}
+	if !bytes.Equal(args.CallerAddr, r.stakeAccessAddr) {
```

Comment:
we might change all these view function checks to a specified address - like viewAddress ? :)

Reply:
in this case it works as the auciton sc does not call the getOwner function anywhere. But we might create something hardcoded for view functions - checking caller to be equal with viewAddress.

Reply:
added TODO